lib/ArtifactoryAPI.js

```diff
@@ -271,4 +272,39 @@ ArtifactoryAPI.prototype.downloadFile = function (repoKey, remotefilePath, desti
   return deferred.promise;
 };
 
+/** Upload Build Information
+ * @param   {object} buildInfo - see build.json {@link https://www.jfrog.com/confluence/display/RTF/Artifactory+REST+API#ArtifactoryRESTAPI-BuildUpload} {@link https://github.com/JFrogDev/build-info#build-info-json-format}
+ * @returns {object} A QPromise to a string containing the result.
+ */
+ArtifactoryAPI.prototype.uploadBuild = function (buildInfo) {
+  var deferred = Q.defer();
+  if(!buildInfo.name || !buildInfo.number) {

```

Comment:
Both number are name seems to be strings, right? Then I guess you could call trim() just to discard that someone is sending a bunch of spaces.